src/Microsoft.Diagnostics.Monitoring.WebApi/MetricsStore.cs

```diff
@@ -147,7 +147,7 @@ private static string GetPrometheusMetric(ICounterPayload metric, out string met
             }
 
             metricValue = value.ToString(CultureInfo.InvariantCulture);
-            return FormattableString.Invariant($"{metric.Provider.Replace(".", string.Empty).ToLowerInvariant()}_{metric.Name.Replace('-', '_')}{unitSuffix}");
+            return FormattableString.Invariant($"{metric.Provider.Replace(".", string.Empty).Replace('-', '_').ToLowerInvariant()}_{metric.Name.Replace('-', '_')}{unitSuffix}");
```

Comment:
nit: For the sake consistency.
`Microsoft.AspNetCore.Hosting`/`connections-duration` ➡️  `microsoftaspnetcorehosting_connections_duration`

`Microsoft-AspNetCore-Server-Kestrel`/`connection-queue-length` ➡️ `microsoftaspnetcoreserverkestrel_connection_queue_length`

```suggestion
            return FormattableString.Invariant($"{metric.Provider.Replace(".", string.Empty).Replace('-', string.Empty).ToLowerInvariant()}_{metric.Name.Replace('-', '_')}{unitSuffix}");
```

Reply:
I'm not against this change, but it does alter the [namespace ](https://prometheus.io/docs/practices/naming/) of all the existing metrics. Anyone who has been consuming metrics before will see brand new ones.

Reply:
The namespace of existing namespaces will not be able to be parsed by Prometheus without making this change. As I mentioned in the bug, my understanding is that it must follow the format:  `It may contain ASCII letters and digits, as well as underscores and colons. It must match the regex [a-zA-Z_:][a-zA-Z0-9_:]*.` from https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

Reply:
May I please ask what the resolution time frame may look like for this? Is it just that I should make the change Sourabh suggested? I see that the tests failed on my change but I'm unable to run them successfully locally without my change as well. I end up seeing errors like the following: ﻿
```
Microsoft.Diagnostics.Monitoring.UnitTests.EndpointInfoSourceTests.ServerSourceAddRemoveSingleConnectionTest
   Source: EndpointInfoSourceTests.cs line 108
   Duration: 5.1 sec

  Message: 
    System.InvalidOperationException : No process is associated with this object.

  Stack Trace: 
    Process.dll:token 0x6000130+0x18
    Process.dll:token 0x60000f9+0x8
    DotNetRunner.ForceClose() line 181
    LoggingRunnerAdapter.DisposeAsync() line 62
    AppRunner.DisposeAsync() line 102
    AppRunnerExtensions.ExecuteAsync(AppRunner runner, Func`1 func) line 44
    EndpointInfoSourceTests.ServerSourceAddRemoveSingleConnectionTest() line 120
    EndpointInfoSourceTests.ServerSourceAddRemoveSingleConnectionTest() line 139
    --- End of stack trace from previous location ---
```

Reply:
@webbsk I am currently working on a change to address #531 which will normalize metric names and providers. I expect to get it out for review in the next day or two.

Reply:
@wiktork That's great news! Thanks for the update! 
May I please ask - will this be available as another preview release, such as tag 6? Or should I switch back to the nightly images based off image 5 for now in order to pick up your changes?

Reply:
About 24 hours after the change is merged (and if all of the builds succeed, but I try to stay on top of that), then it will be available in the nightly image.

However, this upcoming preview will have breaking changes compared to previous previews regarding the HTTP API surface. But if you are only using the `/metrics` route on the 52325 port, then I don't think any of those changes will affect you.